### PR TITLE
Fix method length

### PR DESCRIPTION
**Pull request by EGiataganas**

### CODE REVIEW

**Comment by SteveMarshall** [comment]
@EGiataganas What's happening with this? Is it worth working out the conflicts and getting it reviewed, or shall we just close it?

**Comment by EGiataganas** [comment]
@SteveMarshall let me have a look

**Comment by EGiataganas** [comment]
@dan-palmer @tatyree @SteveMarshall Please have a look!

File: app/controllers/concerns/trim_params.rb [hunk 1]
```diff
@@ -14,16 +14,28 @@ def strip_parameter_whitespace
...
+      array?(obj)
     when String
-      obj.strip
+      string?(obj)
     else
       obj
     end
   end
+
+  def hash?(obj)
```
Comment (dan-palmer) [inline app/controllers/concerns/trim_params.rb]: Feels a little strange to have these look like predicate methods, I'd rename them.

Comment (dan-palmer) [inline app/controllers/concerns/trim_params.rb]: These type specific white space stripping methods should also be made private.

File: app/mailers/prison_mailer.rb [hunk 2]
```diff
@@ -95,6 +87,18 @@ def recipient
 
   private
 
+  def mail_of_booking_request_email(user)
```
Comment (tatyree) [inline app/mailers/prison_mailer.rb]: Could this be renamed `booking_request_email`, or even `booking_request`?

File: app/jobs/zendesk_tickets_job.rb [hunk 3]
```diff
@@ -9,12 +9,16 @@ def perform(feedback)
...
   end
+
+  def custom_fields(feedback)
+    [
+      { id: '23730083', value: feedback.referrer },
+      { id: '23757677', value: 'prison_visits' },
+      { id: '23791776', value: feedback.user_agent },
+      { id: '23984153', value: feedback.prison }
+    ]
+  end
```
Comment (tatyree) [inline app/jobs/zendesk_tickets_job.rb]: I'd prefer if we weren't hardcoding zendesk metadata in an instance  method at all.  Perhaps put it in Rails.config or an initialiser?

File: app/controllers/metrics_controller.rb [hunk 4]
```diff
@@ -90,4 +77,30 @@ def prison_param
...
+      @start_date, @end_date = fortnightly_range.first, fortnightly_range.last
+      @dataset = CalculatedMetrics.new(
+        VisitMetricsEntry.deferred, 3.days, fortnightly_range
+      )
+    end
+  end
+
+  def start_of_year
+    Date.new(year_param, 1, 1) - Date.new(year_param, 1, 1).wday + 1
+  end
```
Comment (threedaymonk) [inline app/controllers/metrics_controller.rb]: I think this whole method can be replaced with

```ruby
Date.commercial(year_param)
```

```ruby
Date.new(2015, 1, 1) - Date.new(2015, 1, 1).wday + 1
# => Mon, 29 Dec 2014
Date.commercial(2015)
# => Mon, 29 Dec 2014
```

Comment (SteveMarshall) [inline app/controllers/metrics_controller.rb]: Oooh, I'd not seen `Date.commercial` before! :bowtie:

Comment (EGiataganas) [inline app/controllers/metrics_controller.rb]: @threedaymonk nice one!! :+1:

**Pull request merged**